### PR TITLE
[UXIT-1765] - Use new Listbox components in FormListbox

### PR DESCRIPTION
**Pull request by CharlyMartin**
## 📝 Description

This PR updates `FormListbox` by re-using the existing `Listbox` components

## 🛠️ Key Changes

The first commit (https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/844/commits/d2647828d3a9c34db17ad11a53f2257b62b7ae69) reworks `FormListbox` and some existing `Listbox` to accommodate for the form design:
- `ListboxButton` now accepts an error state
- `ListboxOptions` now accepts an array of widths

The second commit (de6ab062ac9affcdfcfa003974e032fc68ffa9bc) groups `Listbox` components under the same folder

## 🧪 How to Test

Make sure the `Listbox` components still work and look as expected.

## 🔖 Resources

- https://headlessui.com/react/listbox#setting-the-dropdown-width

### CODE REVIEW

File: src/app/_components/ListboxOptions.tsx [hunk 1]
```diff
@@ -11,20 +11,30 @@ type HeadlessUIAnchorProps = NonNullable<
...
 
+const widthClasses = {
+  auto: 'w-auto',
+  button: 'w-[var(--button-width)]',
+  xs: 'w-[16rem]',
+  sm: 'w-[20rem]',
+  md: 'w-[24rem]',
+  lg: 'w-[28rem]',
+  xl: 'w-[32rem]',
+} as const
```
Comment (mirhamasala) [inline src/app/_components/ListboxOptions.tsx:31]: Let's see if we still can revert this when we use the abbreviations -`Dapp`, `NFTs`, `DAO`. We can open a new PR for the renaming :-)

File: src/app/_components/Form/FormListbox.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        value={value}
+        invalid={Boolean(error)}
+        className={buttonWidth}
+      >
+        <ListboxButton
+          text={value?.name || placeholder}
+          leadingIcon={icon}
+          hasError={Boolean(error)}
+        />
+        <ListboxOptions width={optionsWidth} position={optionsPosition}>
```
Comment (barbaraperic) [inline src/app/_components/Form/FormListbox.tsx:56]: why do we pass `optionsWidth`? 🤔 Also do we want the `listboxOptions` to have the same width as `listboxButton` in the project form? 

Screenshot 2024-11-21 at 17 04 37

File: src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx [hunk 3]
```diff
@@ -119,7 +119,7 @@ export function EcosystemProjectForm({
           placeholder="Select year"
           options={yearOptions}
           disabled={isSubmitting}
-          innerWidth="w-40"
+          buttonWidth="w-40"
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx:122]: Let's see if we can just have it be as wide as the button text.

File: src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx [hunk 4]
```diff
@@ -155,6 +155,8 @@ export function EcosystemProjectForm({
               placeholder="Select topic"
               options={subCategoryOptions}
               disabled={isSubmitting}
+              optionsWidth="lg"
```
Comment (mirhamasala) [inline src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx:158]: This will go.

**Review by barbaraperic — APPROVED** [review]
Looking good! ✨

File: src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx (hunk 3 above)
Comment (CharlyMartin) [reply src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx:122]: It causes the button to collapse when selecting an option, which is not great. So, I think we should stick with the current implementation for now.

https://share.cleanshot.com/YFBs6kYx

File: src/app/_components/Form/FormListbox.tsx (hunk 2 above)
Comment (CharlyMartin) [reply src/app/_components/Form/FormListbox.tsx:56]: Yes, that's a good point. We discussed it with @mirhamasala, and we will make `ListboxOptions` as wide as the content.

So we won't be passing `optionsWidth` anymore :)

File: src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx (hunk 4 above)
Comment (CharlyMartin) [reply src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx]: ✅

File: src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx (hunk 3 above)
Comment (mirhamasala) [reply src/app/ecosystem-explorer/project-form/components/EcosystemProjectForm.tsx:122]: Added to the backlog!

**Review by mirhamasala — APPROVED** [review]
Yay! ✨ LGTM! 🙌🏼

**Pull request merged**